configs/config.yml

```diff
@@ -57,33 +53,27 @@ optim:
     scheduler_args: {"mode":"min", "factor":0.8, "patience":10, "min_lr":0.00001, "threshold":0.0002}
 
 dataset:
-  processed: False # if False, need to preprocessor data and generate .pt file
-  # Whether to use "inmemory" or "large" format for pytorch-geometric dataset. Reccomend inmemory unless the dataset is too large
-  # dataset_type: "inmemory"
-  #Path to data files
+  processed: False
+  # Path to data files
   src: "/global/cfs/projectdirs/m3641/Shared/Materials_datasets/MP_data_npj/raw/"
-  #Path to target file within data_path
+  # Path to target file within data_path
   target_path: "/global/cfs/projectdirs/m3641/Shared/Materials_datasets/MP_data_npj/targets.csv"
-  #Path to save processed data.pt file
+  # Path to save processed data.pt file
   pt_path: "/global/homes/s/shuyijia/datasets/MP_data_npj/"
-  #Format of data files (limit to those supported by ASE)
```

Comment:
Could you also add in the transforms into here? Ideally we'd like to have all possible params listed in config.yml and then only the necessary ones in the examples

Reply:
Sure, this would make things a bit more streamlined. Also in that case, we should not make `GetY` a special exception, since I had it like that before. That way it is always specified in the config to make it clear what transforms are being applied. This should eliminate use of the `target_index` parameter since it's only used for `GetY`. What do you think?

Reply:
For transforms, I also added two modifications:
1. It looks like this was a bug which was not caught earlier. Earlier, the composition of transforms was applied to each data object inside the main loop iterating over `sdict` objects. However, node and edge features are applied after this, making this error-prone. I've modified the code to apply transforms after all preprocessing occurs in a separate loop.
2. Instead of making `otf` a global attribute used to determine on-the-fly application of transforms, I made it a transform-level parameter, so certain transforms can be applied on-the-fly by choice. This allows us to make distinguishments between things like `GetY` and `LineGraphMod`, the former of which needs to be applied during preprocessing.

Reply:
Nice! And yeah I completely agree with including GetY so that it's clear what transforms are applied. I kind of think this also gets to a larger issue with our approach, which is that GetY isn't always needed/wanted. Take the DOS model for example, that doesn't use `y` as the target attribute at all, so it doesn't really make sense to always be doing it. Right now, though, our code will fail if `y` isn't included because we use it, at a minimum, to determine certain properties. I think in the future, we might want to rethink how we approach this, and maybe instead just have a 'target' attribute that could be `y` or anything else that the user wants.

Right now, though, if it's not already in there, we should add something to automatically fail the build if `GetY` isn't included in the config. Is that there already?

Reply:
Sure, I could add a simple assertion at the two different areas where transforms are applied (so in `data.py` and `preprocessor.py`) that fixes this for now.